### PR TITLE
Update method panel when clicking on a method in the usage panel

### PR DESCRIPTION
**Pull request by norascheuch**
When a user clicks on a method in the methods usage panel  this PR populates the method modeling panel with the corresponding method.

## Checklist

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

File: extensions/ql-vscode/src/model-editor/methods-usage/methods-usage-data-provider.ts [hunk 1]
```diff
@@ -165,7 +165,9 @@ export class MethodsUsageDataProvider
 
 export type MethodsUsageTreeViewItem = Method | Usage;
 
-function isExternalApiUsage(item: MethodsUsageTreeViewItem): item is Method {
+export function isExternalApiMethod(
```
Comment (norascheuch) [inline extensions/ql-vscode/src/model-editor/methods-usage/methods-usage-data-provider.ts:168]: We should consider putting this in a shared file for the model-editor.

Comment (charisk) [reply extensions/ql-vscode/src/model-editor/methods-usage/methods-usage-data-provider.ts:168]: Agree. Shall we have a separate PR to move it to `method.ts` and rename it? I'd call it `isMethod` as we're gradually moving away from the term External API.

**Comment by norascheuch** [comment]
At the moment, when clicking on the method we automatically open the first usage. Is this the behaviour we want or should we instead 'preserve' state and reopen on the usage that was formerly selected?

https://github.com/github/vscode-codeql/assets/29482385/6d026524-2057-4c43-b51a-0c1de6456ed6

File: extensions/ql-vscode/src/model-editor/methods-usage/methods-usage-data-provider.ts (hunk 1 above)
Comment (norascheuch) [reply extensions/ql-vscode/src/model-editor/methods-usage/methods-usage-data-provider.ts:168]: Sounds good!

**Pull request closed without merging**